.github/workflows/reusable-release.yml

```diff
@@ -302,6 +303,19 @@ jobs:
           path: ${{ env.LOCAL_ARTIFACTS_STAGING_PATH }}
           key: unsigned-${{ env.RUN_ATTEMPT_UID }}
           fail-on-cache-miss: true
+      - name: Verify artifact hashes before signing
+        run: |
+          sudo apt-get install hashdeep -q > /dev/null
+          ARTIFACT_SHA256SUMS_FILE=$( mktemp )
+          printf "$ARTIFACT_SHA256SUMS" > $ARTIFACT_SHA256SUMS_FILE
+          
+          cd $LOCAL_ARTIFACTS_STAGING_PATH
+          echo "Checking artifact hashes..."
+          if [[ $(sha256deep -r -l -X "$ARTIFACT_SHA256SUMS_FILE" .) ]]
```

Comment:
If `sha256deep -r -l -X` produces any output, then we have a mismatch from the hashes recorded in `ARTIFACT_SHA256SUMS_FILE`.

I originally assumed that the exit code of `sha256deep` would be non-zero if there was a mismatch, but actually it's consistently zero no matter what - this differs from `sha256sum`. I used `sha256deep` because it handles recursively hash directories, unlike `sha256sum`, which doesn't: https://askubuntu.com/q/1091335